libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -925,16 +936,29 @@ class MapboxNavigation(
         @FeedbackEvent.Source feedbackSource: String,
         screenshot: String?,
         feedbackSubType: Array<@FeedbackEvent.SubType String>? = emptyArray(),
+        feedbackMetadata: FeedbackMetadata? = null,
     ) {
         MapboxNavigationTelemetry.postUserFeedback(
             feedbackType,
             description,
             feedbackSource,
             screenshot,
             feedbackSubType,
+            feedbackMetadata,
         )
     }
 
+    /**
+     * Provides wrapper of [FeedbackMetadata]. [FeedbackMetadata] is used to send deferred
+     * feedback attached to passed location. It contain data(like location, time) when method is
+     * invoked.
+     *
+     * Note: method throws [IllegalStateException] if trips session is not
+     * started([startTripSession])
```

Comment:
developers expecting that this method return metadata, I think it's better to throw an exception to be explicitly. Otherwise, flow`MapboxNavigation#provideFeedbackMetadataWrapper` -> `wrapper#get` -> `MapboxNavigation#startTripSession`-> `MapboxNavigation#postUserFeedback`(using metadata) is working fine, but feedback provided for current location. I'd only add to docs that need to check if a session is started via `MapboxNavigation#getTripSessionState`